helix-core/src/main/java/org/apache/helix/HelixAdmin.java

```diff
@@ -381,6 +381,17 @@ void manuallyEnableMaintenanceMode(String clusterName, boolean enabled, String r
    */
   void setClusterManagementMode(ClusterManagementModeRequest request);
 
+  /**
+   * Gets cluster management status {@link ClusterManagementMode}: what mode the cluster is and
+   * whether the cluster has fully reached to that mode.
+   *
+   * @param clusterName cluster name
+   * @return {@link ClusterManagementMode}
+   */
+  default ClusterManagementMode getClusterManagementMode(String clusterName) {
```

Comment:
Why you need a default implementation here? We can make it in real implementation right? It is not necessary.

Reply:
It's for backwards compatibility. Because some users implements the `HelixAdmin` interface for MockHelixAdmin in their code, it would break the build when bumping up the new helix lib.

Reply:
HelixAdmin is something not designed to let user implement. Even if they have different plugin, they should extend our implemented class.

Reply:
Synced with JK.
* we don’t expect users to implement (eg. user apis: HelixAdmin, HelixManager) -> no default
* expect users to implement (eg. StateModel) -> default